src/pages/Statistics/StatisticsPage.tsx

```diff
@@ -1,31 +1,56 @@
 import React, { useContext, useEffect, useState } from 'react'
-import { Tabs, Card } from 'antd'
+import { Tabs, Card, InputNumber } from 'antd'
 import { getStatistics } from '../../services/StatinaApi'
 import { UserContext } from '../../services/userContext'
 import { StatisticsBoxPlot } from '../../components/StatisticsBoxPlot/StatisticsBoxPlot'
 import { StatisticsScatterPlot } from '../../components/StatisticsScatterPlot/StatisticsScatterPlot'
 import { Loading } from '../../components/Loading'
 
+const { TabPane } = Tabs
+
 export function StatisticsPage() {
   const userContext = useContext(UserContext)
   const [statistics, setStatistics] = useState<any>()
   const [selectedPlot, setSelectedPlot] = useState<string | undefined>()
-  const defaultTabKey = 0
   const [isLoading, setIsLoading] = React.useState<boolean>(true)
+  const [numberOfcases, setNumberOfcases] = useState<number>(20)
+  const [showTotal, setShowTotal] = useState<boolean>(false)
+  const defaultTabKey = 0
 
   useEffect(() => {
-    getStatistics(userContext)
+    getStatistics(userContext, numberOfcases)
       .then((response) => {
         setStatistics(response)
         setSelectedPlot(response.box_plots[defaultTabKey])
         setIsLoading(false)
       })
       .catch(() => setIsLoading(false))
   }, [])
+
   const onTabChange = (key: string) => {
     setSelectedPlot(key)
   }
-  const { TabPane } = Tabs
+
+  const onNumberOfBatchesChange = (value: number) => {
+    if (value !== null && value > 0) {
```

Comment:
I suggest using the ramda library for these checks, makes the code more clean. We use it in the other repos. For example we could use isFalsy for this https://char0n.github.io/ramda-adjunct/2.6.0/isFalsy.js.html